openfisca_france/model/prestations/minima_sociaux/ass.py

```diff
@@ -167,6 +167,50 @@ def revenus_tns():
 
         return result
 
+class ass_eligibilite_cumul_individu(Variable):
+    value_type = bool
+    label = u"Eligibilité au cumul de l'ASS avec un revenu d'activité"
+    entity = Individu
+    definition_period = MONTH
+    reference = u"https://www.legifrance.gouv.fr/eli/decret/2017/5/5/ETSD1708117D/jo/article_2"
+
+    def formula_2017_09_01(individu, period):
+        # liste des périodes à contrôler plus la période en cours
+        periods_a_inclure = [
+            period.offset(-12),
+            period.offset(-11),
+            period.offset(-10),
+            period.offset(-9),
+            period.offset(-8),
+            period.offset(-7),
+            period.offset(-6),
+            period.offset(-5),
+            period.offset(-4),
+            period.offset(-3),
+            period.offset(-2),
+            period.offset(-1),
+            period,
+        ]
+
+        nb_mois_cumul = 0
+        nb_mois_consecutif_sans_activite = 0
+
+        for period_a_tester in periods_a_inclure:
```

Comment:
@benjello @fpagnoux auriez-vous des suggestions par rapport à cette implémentation ?

Reply:
- Définir `periods_a_inclure ` avec un `range(-12, 0 + 1 )`
- Extraire le contenu de la boucle dans une sous-fonction ou une variable `mois_cumul`, pour éviter la boucle impérative qui est relativement difficile à lire. À la fin, 
`nb_mois_cumul = sum(mois_cumul(period) for period in periods_a_inclure)` ou 
`nb_mois_cumul = sum(individu('mois_cumul_xxx', period) for period in periods_a_inclure)`

Reply:
Merci @fpagnoux 
Malheureusement, avec ce type de découpage, je ne parviens pas à voir comment je peux conserver `nb_mois_consecutif_sans_activite` entre deux appels.
Ce décompte est mis à jour dans la boucle comme ceci : 
`nb_mois_consecutif_sans_activite = where(absence_ressources_activite * chomeur, nb_mois_consecutif_sans_activite + 1, 0)` 
Or, ce décompte est capital car, comme noté en commentaire dans la formule : 
`# reinitialisation du nombre de mois de cumul après 3 mois consécutif sans activité`

Reply:
@frtomas peux-tu utiliser un `range`  pour les périodes. La boucle est le mieux qu'on puisse faire pour le moment. On la merde dès que possible.

Reply:
@guillet j'ai fait la modif pour utiliser un range au lieu d'une liste de periode